File: 18주차/이연상/이진 변환 반복하기.py
# [PRG] 이진 변환 반복하기

def solution(s):
    answer = [0 , 0]
    
    while(s != "1"):
        answer[1] += s.count('0')
        s = s.replace('0', "")
        s = bin(len(s))
        s = str(s[2:])
        answer[0] += 1
    
    return answer

File: 39주차/이연상/이진 변환 반복하기.java
// [PRG] 이진 변환 반복하기

class Solution {
    public int[] solution(String s) {
        int a = 0; int b = 0;
        while(s != "1") {
            for(int i = 0; i < s.length(); i++) {
                if(s.charAt(i) == '0') {
                    b += 1;
                }
            }
            s = s.replace("0", "");
            if(s.length() == 1) {
                a += 1;
                break;
            }
            s = Integer.toString(s.length(), 2);
            a += 1;
        }
        
        return new int[]{a, b};
    }
}